packages/expo-sqlite/web/WorkerChannel.ts

```diff
@@ -117,13 +117,23 @@ export function invokeWorkerSync<T extends SQLiteWorkerMessageType & keyof Resul
   });
 
   let i = 0;
+  // @ts-expect-error: Remove this when TypeScript supports Atomics.pause
+  const useAtomicsPause = typeof Atomics.pause === 'function';
   while (Atomics.load(lock, 0) === PENDING) {
-    // NOTE(kudo): Unfortunate busy loop,
-    // because we don't have a way for main thread to yield its execution to other callbacks.
-    // Maybe we can wait for [`Atomics.pause`](https://github.com/tc39/proposal-atomics-microwait) to be implemented.
     ++i;
-    if (i > 1000000000) {
-      throw new Error('Sync operation timeout');
+
+    if (useAtomicsPause) {
+      if (i > 1_000_000) {
```

Comment:
The previous value was `1_000_000_000`, is making it smaller intended?

Reply:
yes. `Atomics.pause` will use cpu instruction to pause current thread. compared to the busy loop approach, we can use a lower value for timeout